### ISSUE TITLE
Meaning of "unknown type prefix"?

### ISSUE THREAD

**Issue by markschmid** [mediocregopher/radix#135]
In our application logs we discovered hundreds of errors containing the string ```unknown type prefix``` followed by a single digit string enclosed by single quotation marks, e.g.

```
unknown type prefix '7'
unknown type prefix '9'
unknown type prefix '3'
unknown type prefix '4'
...
```

We've checked in the code your UnmarshalRESP method but don't really understand the meaning of it. Does it mean we have strange data in redis or rather that redis is in a strange state?

Any help kindly appreciated, it's actually occurring a production enviroment :)

**Comment by nussjustin** [comment]
What kind of Client do you use for connecting to radix (Conn, Pool, Sentinel or Cluster)?

Every message Redis sends to the client starts with a prefix denoting what type the message has which in turn tells the client how to parse it. The possible prefixes are `:`, `$`, `+`, `-` and `*`. 

In your case the client sees some unexpected type prefix and returns an error. This means that either the Redis server returns garbage data (extremely unlikely) or that at some point the connection returns an error, but the connection is reused after that.

PR #127 was merged a few hours ago which should help with these cases (there is/was a similar problem in #126, which lead to PR #127).

You could try #127 by updating to the current master or wait until a new release is tagged.

**Comment by mediocregopher** [comment]
Yeah, if you could tell us if you're getting the same behavior with latest
master that would be really helpful, it does sound like the problem that PR
was intended to fix.

**Comment by markschmid** [comment]
We're using a Sentinel client and have noticed that some days ago there was a switch of the master, though we do currently not think it's related with this issue. We're building our (Dockerized) application using

```
go get -d -v github.com/mediocregopher/radix
```

To my slight embarrassment, I'm not really sure if this already gets the current master or the current latest release of your library (v3.3.0).

**Comment by nussjustin** [comment]
> To my slight embarrassment, I'm not really sure if this already gets the current master or the current latest release of your library (v3.3.0).

Are you using Go modules? When using Go modules you would be able to see the version in the `go.mod` file and can update it using `go get -u`. Otherwise `go get` should always fetch the latest master (given that you don't already have the package in your `$GOPATH` in which case `go get` does not update the package unless you pass the `-u` flag).

#127 was merged 2 days ago, so unless you have redeployed since then you don't have the necessary fixes.

> Also, we're experiencing intermittent occurrencies of unexpected responses, e.g.

This can happen in this case too. The same happened in #126.

**Comment by markschmid** [comment]
Great (and fast) responses, thanks!
Unfortunately, we have not switched to Go modules in this project. In any case, we'll try to make sure to get the latest master and see what happens.

**Comment by markschmid** [comment]
Yes indeed, after the deployment of the new version all those errors are gone. It shows that we certainly need to vendor the libraries we're using... Thanks again for the help!

**Comment by nussjustin** [comment]
Good to here.

There are still some open problems with timeouts, etc. that will be addressed in #139. After that is fixed (and maybe #137) I think we should tag a new release. @mediocregopher

**Comment by mediocregopher** [comment]
Definitely we'll tag after #139, can't make promises about #137 being in it (mostly due to my taking forever to review things >_<).

Gonna close this, glad we (mostly justin) could help! :)
